src/main/java/com/sps/Servlets/NewRecipeServlet.java

```diff
@@ -134,13 +136,13 @@ private void addToSearchStrings(Collection<String> searchStrings, String stringT
   private Recipe entityToRecipe(Entity recipeEntity) {
     String name = (String) recipeEntity.getProperty("name");
     String description = (String) recipeEntity.getProperty("description");
-    LinkedList<String> tags = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG);
-    LinkedList<String> ingredients = (LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("ingredients"), INGREDIENT);
+    LinkedHashSet<String> tags = new LinkedHashSet<>((LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("tags"), TAG));
+    LinkedHashSet<String> ingredients = new LinkedHashSet<>((LinkedList<String>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("ingredients"), INGREDIENT));
     LinkedList<Step> steps = (LinkedList<Step>) (LinkedList<?>) getDataAsList(recipeEntity.getProperty("steps"), STEP);
     return new Recipe(name, description, tags, ingredients, steps);
   }
 
-  private LinkedList<Object> getDataAsList(Object propertiesObject, String type) {
+  private Collection<Object> getDataAsList(Object propertiesObject, String type) {
```

Comment:
Why did this change to collection?

Reply:
Previously, tags, ingredients, and steps all used List, but now tags and ingredients use LinkedHashSet, so a Collection ensures that this method can be applied to all of them.